VD_Transmit/VD_Transmit.ino

```diff
@@ -0,0 +1,177 @@
+  uint8_t b[8]; // Adafruit CAN library wants uint8_t[8]
+};
+
+millisDelay CAN_Send_Delay; 
+
+
+void setup()
+{
+  /*******************************************************************************/
+  pinMode(PIN_CAN_STANDBY, OUTPUT);
+  digitalWrite(PIN_CAN_STANDBY, false); // turn off STANDBY
+  pinMode(PIN_CAN_BOOSTEN, OUTPUT);
+  digitalWrite(PIN_CAN_BOOSTEN, true); // turn on booster
+
+  Serial.begin(115200);
+  // start the CAN bus at 500 kbps
+    CAN.begin(500000);
+
+  // creating a millsDelay object using the library <millsDelay.h>
+  // This provides non-blocking delays that don't obstruct execution
+  // like using the standard delay() function would. 
+  CAN_Send_Delay.start(10); // We want to execute after every 1000 milliseconds
+  
+  /*******************************************************************************/
+  bool lsm6ds_success, lis3mdl_success;
+
+  /* Hardware I2C mode, can pass in address & alt wire */
+  lsm6ds_success = lsm6ds.begin_I2C();
+  lis3mdl_success = lis3mdl.begin_I2C();
+
+  if (!lsm6ds_success){
+    //Serial.println("Failed to find LSM6DS chip");
+  }
+  if (!lis3mdl_success){
+    //Serial.println("Failed to find LIS3MDL chip");
+  }
+  
+  //Serial.println("LSM6DS and LIS3MDL Found!");
+  
+  /*******************************************************************************/
+  /* Configure accelerometer range */
+  lsm6ds.setAccelRange(LSM6DS_ACCEL_RANGE_2_G);
+
+  /* Configure accelerometer data rate */
+  lsm6ds.setAccelDataRate(LSM6DS_RATE_208_HZ);
+
+  /* Configure gyro range */
+  lsm6ds.setGyroRange(LSM6DS_GYRO_RANGE_250_DPS);
+
+  /* Configure gyro data rate */
+  lsm6ds.setGyroDataRate(LSM6DS_RATE_208_HZ);
+
+  /* Configure magnetometer data rate */
+  lis3mdl.setDataRate(LIS3MDL_DATARATE_155_HZ);
+  // You can check the datarate by looking at the frequency of the DRDY pin
+
+  /* Configure magnetometer data range */
+  lis3mdl.setRange(LIS3MDL_RANGE_4_GAUSS);
+
+  /* Configure magnetometer performance mode */
+  lis3mdl.setPerformanceMode(LIS3MDL_HIGHMODE);
+
+  /* Configure magnetometer operation mode */
+  lis3mdl.setOperationMode(LIS3MDL_CONTINUOUSMODE);
+
+  lis3mdl.setIntThreshold(500);
+  lis3mdl.configInterrupt(false, false, true, // enable z axis
+                          true, // polarity
+                          false, // don't latch
+                          true); // enabled!
+                          
+  /*******************************************************************************/
+}
+
+void loop()
+{
+  /**********************************************************************************************/
+  /*************************************[BEGIN IMU LOOP]*****************************************/
+  /**********************************************************************************************/
+  sensors_event_t accel, gyro, mag, temp;
+
+  /* Get new normalized sensor events */
+  lsm6ds.getEvent(&accel, &gyro, &temp);
+  lis3mdl.getEvent(&mag);
+
```

Comment:
Since you don't do anything with the data yet unless it's being sent, you can do this operation inside the timed function. 
Potentially you could also do the ```getEvent``` and ```encode_blah_blah``` calls into their own function which would clean up your loop code. 
No need at this point though. 

If you want to read it on every cycle or the purposes of sending out filtered data at some point, just leave it here.  Your filter variables will have to exist outside of the loop though.

Reply:
I'll definitely clean this code up in the future just for the sake of making it more legible. As I said in a previous reply, I like to organize my code to spare myself future headaches.

I also didn't realize that I wasn't making use of the timer. I'll fix this in a future iteration.